lib/data/data_source/remote/folders_data_source_impl.dart

```diff
@@ -15,13 +15,19 @@ class FoldersDataSourceImpl implements FoldersDataSource {
 
   final FirebaseFirestore firestore;
   final FirebaseAuth firebaseAuth;
+  String users = 'users';
 
   String? get userId => firebaseAuth.currentUser?.uid;
 
   @override
-  Future<Success> newFolder(FolderDto dto) async {
+  Future<Success> createCollection(FolderDto dto) async {
     if (userId != null) {
-      await firestore.collection("$userId").doc(dto.folderName.toUpperCase()).set(dto.toJson());
+      await firestore
+          .collection('collections')
+          .doc(userId)
+          .collection('userCollections')
+          .doc(dto.folderName.toUpperCase())
+          .set(dto.toJson());
```

Comment:
this is not compatible with that what you made in the firebase, rethink the models